sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionRulesDefinition.java

```diff
@@ -55,29 +50,29 @@
             "/com/github/sevntu/checkstyle/sonar/checkstyle-extensions.xml";
 
     /**
-     * XML Parser.
+     * XML loader for rules definition.
      */
-    private final XMLRuleParser xmlRuleParser;
+    private final RulesDefinitionXmlLoader xmlRuleLoader;
 
     /**
      * Useless JavaDoc for a Constructor.
-     * @param xmlRuleParser obviously the XML Parser as it already says, what else?!
+     * @param xmlRuleLoader rules definition xml loader.
      */
-    public CheckstyleExtensionRepository(XMLRuleParser xmlRuleParser) {
-        super(REPOSITORY_KEY, REPOSITORY_LANGUAGE);
-        setName(REPOSITORY_NAME);
-        this.xmlRuleParser = xmlRuleParser;
+    public CheckstyleExtensionRulesDefinition(RulesDefinitionXmlLoader xmlRuleLoader) {
+        this.xmlRuleLoader = xmlRuleLoader;
     }
 
-    @Override
-    public List<Rule> createRules() {
-        final InputStream input = getClass().getResourceAsStream(RULES_RELATIVE_FILE_PATH);
-        try {
-            return xmlRuleParser.parse(input);
-        }
-        finally {
-            IOUtils.closeQuietly(input);
-        }
-    }
+    /**
+     * {@inheritDoc}
+     */
+    public void define(Context context) {
```

Comment:
From the javadoc, this should have `@Override`? Why isn't it listed here with no javadoc.

@romani Shouldn't one of our checks have caught this?

Reply:
@rdiachenko I would check that checkstyle is running on all files.

Reply:
If that is simple to fix, please do, but it is not related problem it is better to address it in separate issue/PR.

Reply:
@rnveach , sonar-sevntu-checkstyle uses this checkstyle configuration for validation: https://raw.githubusercontent.com/checkstyle/checkstyle/checkstyle-8.18/config/checkstyle_checks.xml
Here's config for **MissingOverride** module:
```
    <module name="MissingOverride">
      <property name="javaFiveCompatibility" value="true"/>
    </module>
```
I assume the validation passes because of **javaFiveCompatibility=true**.

Reply:
looks like we need to disable this property :)

Reply:
Fixed in a separate commit - https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/788/commits/79f1d85fe932330875b108bb08249c419d483721